File: 2023/May/week1/정환훈/1_개인정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int dtoi(string date){
    string year = date.substr(0, 4);
    string month = date.substr(5, 2);
    string day = date.substr(8);
    return stoi(year)*12*28 + stoi(month)*28 + stoi(day);
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;
    map<char, int> m;
    for(auto term : terms){
        m[term[0]] = stoi(term.substr(2));
    }
    int todayI = dtoi(today);
    for(int j=0;j<privacies.size();j++){
        string p = privacies[j];
        string date = p.substr(0, 10);
        int i = dtoi(date) + m[p[11]] * 28;
        if(todayI >= i) answer.push_back(j+1);
    }
    return answer;
}

File: 2023/May/week1/이정욱/1_개인정보 수집 유효기간.py
def getDate(day):
    day = list(map(int,day.split('.')))
    return day[0] * 12 * 28 + day[1] * 28 + day[2]

def solution(today, terms, privacies):
    answer = []
    _terms = {}
    today = getDate(today)
    # print(f"today : {today}")
    
    for i in range(len(terms)):
        e = terms[i].split()
        _terms[e[0]] = int(e[1]) * 28
    
    for i in range(len(privacies)):
        e = privacies[i].split()
        p = getDate(e[0]) + _terms[e[1]]
        
        if p <= today:
            answer.append(i+1)
    return answer